recipes-kernel/linux/linux-yocto/0051-Add-ACPI-support-for-MCP251x-CAN-controller.patch

```diff
@@ -0,0 +1,105 @@
+From 75798138dfe56a092909dbb91be469949fcd2661 Mon Sep 17 00:00:00 2001
+From: Sugnan Prabhu S <sugnan.prabhu.s@intel.com>
+Date: Wed, 20 Sep 2017 10:54:32 +0530
+Subject: [PATCH] Add ACPI support for MCP251x CAN controller
+
+This patch adds ACPI support for mcp251x can controller.
+
+Signed-off-by: Sugnan Prabhu S <sugnan.prabhu.s@intel.com>
+---
+ drivers/net/can/spi/mcp251x.c | 28 ++++++++++++++++++++++++++++
+ 1 file changed, 28 insertions(+)
+
+diff --git a/drivers/net/can/spi/mcp251x.c b/drivers/net/can/spi/mcp251x.c
+index 575790e..68e4c32 100644
+--- a/drivers/net/can/spi/mcp251x.c
++++ b/drivers/net/can/spi/mcp251x.c
+@@ -54,6 +54,7 @@
+  *
+  */
+ 
++#include <linux/acpi.h>
+ #include <linux/can/core.h>
+ #include <linux/can/dev.h>
+ #include <linux/can/led.h>
+@@ -219,6 +220,8 @@
+ 
+ #define DEVICE_NAME "mcp251x"
+ 
++#define DEFAULT_OSC_CLK_FREQ	(25 * 1000 * 1000)
++
+ static int mcp251x_enable_dma; /* Enable SPI DMA. Default: 0 (Off) */
+ module_param(mcp251x_enable_dma, int, S_IRUGO);
+ MODULE_PARM_DESC(mcp251x_enable_dma, "Enable SPI DMA. Default: 0 (Off)");
+@@ -1023,6 +1026,12 @@ static const struct spi_device_id mcp251x_id_table[] = {
+ };
+ MODULE_DEVICE_TABLE(spi, mcp251x_id_table);
+ 
++static const struct acpi_device_id mcp251x_acpi_match[] = {
++	{ "MCHP2515", CAN_MCP251X_MCP2515},
++	{ },
++};
++MODULE_DEVICE_TABLE(acpi, mcp251x_acpi_match);
++
+ static int mcp251x_can_probe(struct spi_device *spi)
+ {
+ 	const struct of_device_id *of_id = of_match_device(mcp251x_of_match,
+@@ -1032,11 +1041,23 @@ static int mcp251x_can_probe(struct spi_device *spi)
+ 	struct mcp251x_priv *priv;
+ 	struct clk *clk;
+ 	int freq, ret;
++	const struct acpi_device_id *id;
++
++	if (has_acpi_companion(&spi->dev)) {
++		id = acpi_match_device(mcp251x_acpi_match, &spi->dev);
++		if (!id)
++			return -EINVAL;
++		spi->irq = acpi_dev_gpio_irq_get(ACPI_COMPANION(&spi->dev), 0);
++		if (spi->irq < 0)
++			return -EINVAL;
++	}
+ 
+ 	clk = devm_clk_get(&spi->dev, NULL);
+ 	if (IS_ERR(clk)) {
+ 		if (pdata)
+ 			freq = pdata->oscillator_frequency;
++		else if (has_acpi_companion(&spi->dev))
++			freq = DEFAULT_OSC_CLK_FREQ;
+ 		else
+ 			return PTR_ERR(clk);
+ 	} else {
+@@ -1067,10 +1088,14 @@ static int mcp251x_can_probe(struct spi_device *spi)
+ 	priv->can.clock.freq = freq / 2;
+ 	priv->can.ctrlmode_supported = CAN_CTRLMODE_3_SAMPLES |
+ 		CAN_CTRLMODE_LOOPBACK | CAN_CTRLMODE_LISTENONLY;
++
+ 	if (of_id)
+ 		priv->model = (enum mcp251x_model)of_id->data;
++	else if (has_acpi_companion(&spi->dev))
++		priv->model = id->driver_data;
+ 	else
+ 		priv->model = spi_get_device_id(spi)->driver_data;
++
+ 	priv->net = net;
+ 	priv->clk = clk;
+ 
+@@ -1155,6 +1180,8 @@ static int mcp251x_can_probe(struct spi_device *spi)
+ 
+ 	devm_can_led_init(net);
+ 
++	dev_info(&spi->dev, "CAN driver registration successful\n");
```

Comment:
this is not part of a "add acpi" patch

Reply:
Shall I make this part of the hardcode patch? Sometimes even though the driver is loaded, can network interface will not be registered?